### PR TITLE
review processing of DeviceRequest due to Docker engine changes

### PR DESCRIPTION
**Pull request by glours**
See https://github.com/moby/moby/pull/48483

Fixes https://github.com/docker/docs/issues/21026

### CODE REVIEW

**Review by ndeloof — APPROVED** [review]

**Review by laurazard — COMMENTED** [review]
LGTM! Thanks

File: loader/loader_test.go [hunk 1]
```diff
@@ -2173,6 +2190,40 @@ services:
...
+    deploy:
+      resources:
+        reservations:
+          devices:
+            - driver: nvidia
+              capabilities: [gpu]
+              count: 2
+              device_ids: ["my-device-id"]
+`)
+	assert.ErrorContains(t, err, `invalid "count" and "device_ids" are attributes are exclusive`)
```
Comment (jhrotko) [inline loader/loader_test.go:2208]:
```suggestion
	assert.ErrorContains(t, err, `invalid "count" and "device_ids", these attributes are exclusive`)
```

Comment (jhrotko) [inline loader/loader_test.go:2208]: Question: why can't `capabilities` and `device_ids` be defined at the same time?

File: types/device.go [hunk 2]
```diff
@@ -50,3 +50,48 @@ func (c *DeviceCount) DecodeMapstructure(value interface{}) error {
...
+	d.Count = DeviceCount(-1)
+
+	capabilities := v["capabilities"]
+	caps := StringList{}
+	if err := caps.DecodeMapstructure(capabilities); err != nil {
+		return err
+	}
+	d.Capabilities = caps
+	if driver, ok := v["driver"]; ok {
+		if val, ok := driver.(string); ok {
```
Comment (jhrotko) [inline types/device.go:76]: shouldn't JSON schema handle this?

**Pull request merged**